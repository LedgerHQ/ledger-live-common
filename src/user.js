// @flow
import { getEnv } from "./env";
import { sha256 } from "./crypto";

const userHashesPerUserId = (userId: string) => {
  const firmwareSalt = sha256(userId + "|firmwareSalt")
    .toString("hex")
    .slice(0, 6);
  return { firmwareSalt };
};

let cache;
export const getUserHashes = () => {
  const userId = getEnv("USER_ID");
  if (cache && userId === cache.userId) {
    return cache.value;
  }
  cache = {
    userId,
    value: userHashesPerUserId(userId),
  };
  return cache.value;
};
