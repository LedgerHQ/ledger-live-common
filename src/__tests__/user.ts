import { getUserHashes } from "../user";
test("stable user", () => {
  expect(getUserHashes()).toMatchSnapshot();
});
