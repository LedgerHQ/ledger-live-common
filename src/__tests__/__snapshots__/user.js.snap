// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stable user 1`] = `
Object {
  "endpointOverrides100": 20,
  "firmwareSalt": "544d89",
}
`;
